### PR TITLE
Add WasmPlugin to scoped config types

### PR DESCRIPTION
**Pull request by dddddai**
**Please provide a description of this PR:**
to reduce unnecessary LDS/ECDS pushes

### CODE REVIEW

**Comment by dddddai** [comment]
Maybe we can filter more granularly by workloadSelector for these types

**Comment by hzxuzhonghu** [comment]
can you elaborate why it can reduce pushes on lds and ECDS?

**Review by hzxuzhonghu — COMMENTED** [review]
Please also add tests for influence on xds push

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
@hzxuzhonghu can you remove hold if it looks good to you?

cc @ingwonsong

**Comment by dddddai** [comment]
> can you elaborate why it can reduce pushes on lds and ECDS?

We don't need to push xDS if updated WasmPlugins are in different namespaces from the proxy

**Comment by hzxuzhonghu** [comment]
@howardjohn Yes, i will take a look. It is lack of description, so was hard to review

File: pilot/pkg/model/sidecar.go [hunk 1]
```diff
@@ -44,12 +44,13 @@ var (
 		kind.Sidecar:         {},
 	}
 
-	// clusterScopedConfigTypes includes configs when they are in root namespace,
+	// namespaceScopedConfigTypes includes configs when they are in root namespace,
 	// they will be applied to all namespaces within the cluster.
-	clusterScopedConfigTypes = map[kind.Kind]struct{}{
+	namespaceScopedConfigTypes = map[kind.Kind]struct{}{
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar.go:49]: revert, i think this is misleading

**Review by hzxuzhonghu — COMMENTED** [review]
I am reading this PR is triggering a push when a WasmPlugin in root namespace updates.

**Comment by dddddai** [comment]
> I am reading this PR is triggering a push when a WasmPlugin in root namespace updates.

Isn't it expected?

**Comment by hzxuzhonghu** [comment]
Yes, this is right.

I mean i am confusing about, actually if the wasmplugin is rootnamespace, we still need to push whenere where the prox y is


>>can you elaborate why it can reduce pushes on lds and ECDS?

>We don't need to push xDS if updated WasmPlugins are in different namespaces from the proxy

**Comment by dddddai** [comment]
Yeah the root namespace is an exception, just like what we are doing for EnvoyFilter etc

**Pull request merged**